contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -662,181 +619,137 @@ contract LiquidityProtection is ILiquidityProtection, Utils, Owned, ReentrancyGu
-                removedPos.reserveRateN,
-                removedPos.reserveRateD,
-                removeSpotRate,
-                removeAverageRate
-            );
-
-            // get the fully protected amount (+ fees)
-            fullyProtectedAmount = fullyProtectedAmount.add(
-                _removeLiquidityTargetAmount(
-                    poolRate,
-                    removedPos.poolAmount,
-                    removedPos.reserveAmount,
-                    packedRates,
-                    Fraction({ n: 1, d: 1 })
-                )
-            );
-        }
-
-        // add the pool tokens to the system
-        _systemStore.incSystemBalance(poolToken, poolTokenAmount);
-
-        // remove network token liquidity
+        // for the network token, return the target amount
         if (_isNetworkToken(reserveToken)) {
-            // mint the fully protected amount (+ fees) and migrate it
-            _mintNetworkTokens(address(this), poolToken, fullyProtectedAmount);
-
-            _networkToken.approve(address(_networkV3), fullyProtectedAmount);
-
-            _networkV3.migrateLiquidity(
-                IReserveToken(address(_networkToken)),
-                msg.sender,
-                fullyProtectedAmount,
-                fullyProtectedAmount,
-                originalAmount
-            );
-
-            return;
+            return (targetAmount, targetAmount);
         }
 
-        // remove base token liquidity
+        Fraction memory removeAverageRate = Fraction({
+            n: packedRates.removeAverageRateN,
+            d: packedRates.removeAverageRateD
+        });
 
-        // calculate the amount of pool tokens required for liquidation
-        // note that the amount is doubled since it's not possible to liquidate one reserve only
-        uint256 poolLiquidationAmount = _liquidationAmount(fullyProtectedAmount, poolRate, poolToken, 0);
+        // calculate the position impermanent loss
+        Fraction memory loss = _impLoss(addSpotRate, removeAverageRate);
 
-        // withdraw the pool tokens from the wallet
-        _withdrawPoolTokens(poolToken, poolLiquidationAmount);
+        // deduct the position IL from the target amount
+        targetAmount = _deductIL(Math.max(reserveAmount, targetAmount), loss);
 
-        // remove liquidity
-        _removeLiquidity(poolToken, poolLiquidationAmount, reserveToken, IReserveToken(address(_networkToken)));
+        // get the pool deficit
+        Fraction memory poolDeficit = _poolDeficit(poolToken);
 
-        // migrate the received tokens
-        uint256 removedAmount = reserveToken.balanceOf(address(this));
-        uint256 value;
-        if (reserveToken.isNativeToken()) {
-            value = removedAmount;
-        } else {
-            IERC20(address(reserveToken)).safeApprove(address(_networkV3), removedAmount);
+        // no deficit
+        if (poolDeficit.n == 0) {
+            return (targetAmount, targetAmount);
```

Comment:
Already handled correctly in the final `return` statement.

Reply:
removed